services/marketplace/client.go

```diff
@@ -79,7 +80,7 @@ func (c *Client) GetPlugin(filter *model.MarketplacePluginFilter, pluginVersion
 // closeBody ensures the Body of an http.Response is properly closed.
 func closeBody(r *http.Response) {
 	if r.Body != nil {
-		_, _ = ioutil.ReadAll(r.Body)
+		_, _ = io.Copy(ioutil.Discard, r.Body)
```

Comment:
What is the difference between discarding and reading into `_`? I would assume that "it does the same".

Reply:
`io.Copy(ioutil.Discard, r.Body)` discards without allocating. `ReadAll` allocates a buffer.

Reply:
Yes, please see: https://github.com/mattermost/mattermost-server/pull/14521#issue-415524611